plugin-viewport/src/main/kotlin/com/mapbox/maps/plugin/viewport/ViewportPluginImpl.kt

```diff
@@ -0,0 +1,389 @@
+import com.mapbox.maps.plugin.viewport.state.ViewportCameraState.TransitionToFollowing
+import com.mapbox.maps.plugin.viewport.state.ViewportCameraState.TransitionToOverview
+import com.mapbox.maps.plugin.viewport.state.ViewportCameraStateChangedObserver
+import com.mapbox.maps.plugin.viewport.transition.MapboxViewportCameraStateTransition
+import com.mapbox.maps.plugin.viewport.transition.TransitionEndListener
+import com.mapbox.maps.plugin.viewport.transition.ViewportCameraStateTransition
+import com.mapbox.maps.plugin.viewport.transition.ViewportCameraTransition
+import com.mapbox.maps.plugin.viewport.transition.ViewportCameraTransitionOptions
+import java.util.concurrent.CopyOnWriteArraySet
+
+/**
+ * Mapbox default implementation for Viewport plugin.
+ */
+class ViewportPluginImpl : ViewportPlugin {
+  private var frameTransitionOptions = DEFAULT_FRAME_TRANSITION_OPT
+  private var runningAnimation: AnimatorSet? = null
+  private val transitionEndListeners = CopyOnWriteArraySet<TransitionEndListener>()
+  private val viewportCameraStateChangedObservers =
+    CopyOnWriteArraySet<ViewportCameraStateChangedObserver>()
+  private lateinit var cameraPlugin: CameraAnimationsPlugin
+  private lateinit var locationComponentPlugin: LocationComponentPlugin
+  private lateinit var mapDelegateProvider: MapDelegateProvider
+  private lateinit var stateTransition: ViewportCameraStateTransition
+
+  /**
+   * Describes an object that provides desired camera positions to [ViewportPlugin].
+   */
+  override lateinit var viewportDataSource: ViewportDataSource
+
+  /**
+   * Returns current [ViewportCameraTransition].
+   * @see registerViewportCameraStateChangedObserver
+   */
+  override var state: ViewportCameraState = Idle
+    private set(value) {
+      if (value != field) {
+        field = value
+        viewportCameraStateChangedObservers.forEach {
+          it.onViewportCameraStateChanged(value)
+        }
+      }
+    }
+
+  private val sourceUpdateObserver = ViewportDataSourceUpdateObserver(::updateFrame)
+
+  /**
+   * Executes a transition to [Following] state. When started, goes to [TransitionToFollowing]
+   * and to the final [Following] when ended. If transition is canceled, state goes to [Idle].
+   *
+   * The target camera position is obtained with [ViewportDataSource.getViewportData].
+   *
+   * @param stateTransitionOptionsBlock options that impact the transition animation from
+   * the current state to the requested state.
+   * Defaults to [ViewportCameraTransitionOptions.maxDurationMs] equal to 3500 millis.
+   * @param frameTransitionOptionsBlock options that impact the transition animations between
+   * viewport frames in the selected state.
+   * This refers to camera transition on each [ViewportDataSource] update when [Following] is engaged.
+   * Defaults to [ViewportCameraTransitionOptions.maxDurationMs] equal to 1000 millis.
+   * @param transitionEndListener invoked when transition ends.
+   */
+  @JvmOverloads
```

Comment:
why overloads? do not see default parameters

Reply:
applies to all methods below

Reply:
The story is that these methods had default value, and these default values is moved to the interface during the plugin interface extraction, and interface can not use `JvmOverloads` annotation 🙃 

I will remove these annotations for now, and maybe we need to manually overload these functions later.